experimental/unitsconversion/src/measureunit.rs

```diff
@@ -0,0 +1,243 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use zerotrie::ZeroTrie;
+use zerovec::ZeroVec;
+
+use crate::{
+    provider::{Base, MeasureUnitItem, SiPrefix},
+    ConversionError,
+};
+
+// TODO(#4369): split this struct to two structs: MeasureUnitParser for parsing the identifier and MeasureUnit to represent the unit.
+pub struct MeasureUnit {
+    /// Contains the processed units.
+    pub contained_units: Vec<MeasureUnitItem>,
```

Comment:
Suggestion: this should be either `SmallVec` (as in fixed_decimal) or `ShortBoxSlice` (as in icu_locid). The common case shouldn't need to allocate a `Vec`.

Reply:
done